### ISSUE TITLE
Combobox is not filtering certain values

### ISSUE THREAD

**Issue by tresh11178** [Esri/calcite-components#4348]
### Actual Behavior

When inputting certain values, the Combobox dropdown does not properly filter the options as expected.

### Expected Behavior

The expectation is that all values should be filtered properly based on the input from the user.

### Reproduction Sample

https://codepen.io/tre11178/pen/KKZXMYW?editors=100

### Reproduction Steps

1. Click into the Combobox
2. Type "F" into the input.
3. Notice that the dropdown is not filtering all options other than "Fire Safety Team".
4. Type "Fi" into the input.
5. Notice the dropdown is still not filtering.'
6. Type "Fir" into the input.
7. Notice that the dropdown properly filters.

### Reproduction Version

beta.80

**Comment by benelan** [comment]
[Implementer Notes]: This is probably because the `forIn` function "Iterates over own and inherited enumerable properties of an object invoking iteratee for each property."

All of the combobox-items are matching the word "define"

**Comment by geospatialem** [comment]
@Elijbet I'm seeing some filtering issues with a few letters on `next`. Put together an [updated codepen](https://codepen.io/geospatialem/pen/oNpOEwa) for reference. Can you take a peek 👀? 

*Note: Added an additional item to showcase filtering further, but tested across the alphabet.*

Some of the filtering experienced:
- d, For results must type "di"
- e, For results must type "edi"
- f, For results must type "fir"
- n, For results must type "na"
- u, For results must type "ur"

![filter results from 'd'](https://user-images.githubusercontent.com/5023024/164799587-c50d8e18-8aea-4054-ae90-64f9375d22ef.png)

![filter results from 'di'](https://user-images.githubusercontent.com/5023024/164799613-19dae15d-675c-4869-8cb3-2f8adc08b323.png)

**Comment by Elijbet** [comment]
Sure, I'll check now.

**Comment by benelan** [comment]
Yeah that's what I thought. Its not just the guid, its the inherited properties too. The first solution worked IIRC.

**Comment by benelan** [comment]
Or if I'm wrong and the `const inputSubset = (({ label, value }) => ({ label, value }))(input as any);` didn't work, I'd look into [`Object.hasOwn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn).

**Comment by Elijbet** [comment]
> Yeah that's what I thought. Its not just the guid, its the inherited properties too. The first solution worked IIRC.

It's the right thing to do, to remove the guid there. But I should still only filter the subset of the input object. And I'll try your suggestion as well @benelan

**Comment by benelan** [comment]
Yeah agreed, we don't need the guid there. I was just referring to when I [requested changes](https://github.com/Esri/calcite-components/pull/4405#pullrequestreview-947577413). That's my fault, I should have been clearer in my [note above](https://github.com/Esri/calcite-components/issues/4348#issuecomment-1095551253) about the __inherited__ enumerable properties being the issue.

**Comment by geospatialem** [comment]
Going to reallocate this on back to `in development`. Thanks for the background/insights @benelan and for investigating, @Elijbet!

**Comment by geospatialem** [comment]
Verified on next ✔️ 

![verification confirmation with 'f' for fire filtering](https://user-images.githubusercontent.com/5023024/167163421-e7465d7f-d1e6-46be-9c4a-a5e696de6b1a.png)
